### DISCUSSION TITLE
Why can't you register multiple event listeners anymore?

### DISCUSSION THREAD

**Discussion by edalzell** [laravel/framework#51564] [General]
I just upgraded an L10 site to L11 via Laravel Shift and my event registration broke because I had previously had multiple listeners on an event.

Any one know why that feature was removed in L11?

I'd PR a "fix" but I suspect it would be closed, so I'd love to understand why that was done.

**Comment by votkapower** [comment]
Also When you use explisit binding to ensure execution order like:
`Event::listen(Event::class, Listener1::class); Event::listen(Event::class, Listener2::class)` 
- when you try `php artisan event:list` and you will see duplicate listeners - once from the laravels auto-linking and once for the manual linking..  pretty sad

**Reply by zawilliams** [reply]
Running into this same issue now. Searching for solution. Was hoping to find something like how middleware sort order works: https://laravel.com/docs/11.x/middleware#sorting-middleware

Seems like #53533 was the open issue that got closed.

I think what I might need to do is manually register some and let some be auto discovery in a different directory.

**Comment by ghost** [comment]
To give you more work on migrating it @edalzell

**Comment by webdevhayes** [comment]
This is fairly easy to recreate laravel 10's event service provider in laravel just by doing it like this :) @edalzell  @zawilliams just add this to app service provider.

```
 /**
     * Bootstrap any application services.
     */
    public function boot(): void
    {
        Vite::prefetch(concurrency: 3);

        $listen = [
            NewJobCreatedEvent::class => [
                EmployerNewJobCreatedListener::class,
                RecruiterJobCreatedListener::class,
            ],

        ];

        foreach($listen as $event => $listeners) {

            foreach($listeners as $listener) {
                Event::listen($event, [$listener, 'handle']);
            }
        }
    }
```

To stop duplicate just make sure you are not type hinting the listener with the event class. so it would just be ($event) rather than (EventClass $event)
